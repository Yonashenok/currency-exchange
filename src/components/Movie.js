import { Link } from 'react-router-dom';
import propTypes from 'prop-types';
import classes from './Movie.module.css';
import RightArrow from '../assets/icon/RightArrow';

function Movie({ movie }) {
  return (
    <Link to={`/movie/${movie.imdbId}`}>
      <div className={classes.container}>
        <div className={classes.icon}>
          <RightArrow />
        </div>
        <img className={classes.img} src={movie.image} alt={movie.title} />
        <div className={classes.info}>
          <p className={classes.title}>{movie.title}</p>
          <p className={classes.year}>
            {`${movie.releaseDate} - ${movie.runTime}min`}
          </p>
          <p className={classes.rating}>{`Rating / ${movie.rating}`}</p>
          <button className={classes.movieBtn} type="button">
            {movie.genre}
          </button>
        </div>
      </div>
    </Link>
  );
}

Movie.defaultProps = {
  movie: {
    imdbId: '',
    image: '',
    title: '',
    releaseDate: '',
    genre: '',
    id: '',
    rating: 5,
    runTime: 60,
  },
};

Movie.propTypes = {
  movie: propTypes.shape({
    imdbId: propTypes.string,
    image: propTypes.string,
    title: propTypes.string,
    releaseDate: propTypes.string,
    genre: propTypes.string,
    id: propTypes.string,
    rating: propTypes.number,
    runTime: propTypes.number,
  }),
};

export default Movie;
