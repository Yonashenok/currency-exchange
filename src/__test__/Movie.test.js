import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Movie from '../components/Movie';

describe('testing Movie.js', () => {
  it('should display movie form the data', () => {
    const mockMovieData = {
      poster: '/',
      title: 'the big bang theory',
      releaseDate: '2007-01-11',
      movieType: 'comedy',
      id: 'tt53738',
      rating: 8,
      runTime: 30,
    };

    render(
      <BrowserRouter>
        <Movie movie={mockMovieData} />
      </BrowserRouter>,
    );

    const title = screen.getByText('the big bang theory');
    expect(title).toBeInTheDocument();
  });

  it('should display movie form the data', () => {
    const mockMovieData = {
      poster: '/',
      title: 'the big bang theory',
      releaseDate: '2007-01-11',
      movieType: 'comedy',
      id: 'tt53738',
      rating: 8,
      runTime: 30,
    };

    const Component = render(
      <BrowserRouter>
        <Movie movie={mockMovieData} />
      </BrowserRouter>,
    );

    expect(Component).toMatchSnapshot();
  });
});
