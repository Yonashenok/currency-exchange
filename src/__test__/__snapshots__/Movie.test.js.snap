// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Movie.js should display movie form the data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/movie/undefined"
      >
        <div
          class="container"
        >
          <div
            class="icon"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              height="24px"
              stroke="#fff"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <img
            alt="the big bang theory"
            class="img"
          />
          <div
            class="info"
          >
            <p
              class="title"
            >
              the big bang theory
            </p>
            <p
              class="year"
            >
              2007-01-11 - 30min
            </p>
            <p
              class="rating"
            >
              Rating / 8
            </p>
            <button
              class="movieBtn"
              type="button"
            />
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/movie/undefined"
    >
      <div
        class="container"
      >
        <div
          class="icon"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            height="24px"
            stroke="#fff"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.75 15l3-3m0 0l-3-3m3 3h-7.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <img
          alt="the big bang theory"
          class="img"
        />
        <div
          class="info"
        >
          <p
            class="title"
          >
            the big bang theory
          </p>
          <p
            class="year"
          >
            2007-01-11 - 30min
          </p>
          <p
            class="rating"
          >
            Rating / 8
          </p>
          <button
            class="movieBtn"
            type="button"
          />
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
